src/components/signup/SignupForm.react.js

```diff
@@ -0,0 +1,83 @@
+// @flow
+import React from 'react'
+import TextField from 'material-ui/TextField'
+import Paper from 'material-ui/Paper'
+import RaisedButton from 'material-ui/RaisedButton'
+import { reduxForm } from 'redux-form'
+
+const SignupForm = (props: Object) => {
+  const {
+    fields: {
+      firstName,
+      lastName,
+      password,
+      email,
+    },
+    signupSubmit,
+  }: {
+    fields: Object,
+    signupSubmit: Function,
+  } = props
```

Comment:
This is what the flow check looks like for an object. It ensures that fields is an Object and signupSubmit is a func